packages/core/src/storage/FileSystem.ts

```diff
@@ -1,8 +1,34 @@
+import type { WalletStorageConfig } from 'indy-sdk'
+
 export interface FileSystem {
   readonly basePath: string
 
   exists(path: string): Promise<boolean>
   write(path: string, data: string): Promise<void>
   read(path: string): Promise<string>
   downloadToFile(url: string, path: string): Promise<void>
+  loadPostgresPlugin?(storageConfig: WalletStorageConfig, storageCreds: WalletStorageCreds): Promise<boolean>
+}
+
+export enum WalletScheme {
+  DatabasePerWallet = 'DatabasePerWallet',
+  MultiWalletSingleTable = 'MultiWalletSingleTable',
+  MultiWalletSingleTableSharedPool = 'MultiWalletSingleTableSharedPool',
+}
+
+export interface StorageConfig {
+  url: string
+  wallet_scheme?: WalletScheme
+  path?: string | undefined
+}
+
+export interface WalletStorageCreds {
+  [key: string]: unknown
+}
+
+export interface StorageCreds {
+  account: string
+  password: string
+  admin_account: string
+  admin_password: string
 }
```

Comment:
The postgres plugin is related to the environment and the wallet type (in this case indy-sdk). I'm not sure the file system is the best place to add this, mainly because the filesystem stays the same if we support something different than indy-sdk.


Maybe we can keep it a bit more generic. Maybe later we can integrate more, but for now maybe can approach it like this:

```ts
import { Agent } from '@aries-framework/core'
import { agentDependencies, loadPostgresPlugin } from '@aries-framework/node'

async function run() {
  const storageConfig = {}
  const storageCreds = {}

  await loadPostgresPlugin(storageConfig, storageCreds)

  const agent = new Agent({
    walletConfig: {
      id: 'id',
      key: 'key',
      storageConfig,
      storageCreds,
      storageType: 'postgres'
  })
}

run()
```